### PR TITLE
create monitoring and health check port.

### PR DESCRIPTION
**Pull request by larry-aptos**
Locally tested, looks good.

image

image

image

Keep alive check: 
image

### CODE REVIEW

File: python/utils/transactions_processor.py [hunk 1]
```diff
@@ -33,6 +38,25 @@ def __init__(
...
+        # Start the Prometheus client; this is a non-blocking call.
+        start_http_server(self.config.monitoring_port)
+
+        # Start the health server.
+        def start_health_server() -> None:
+            class Handler(http.server.SimpleHTTPRequestHandler):
+                def do_GET(self):
+                    self.send_response(http.HTTPStatus.OK)
+                    self.end_headers()
+                    self.wfile.write(b"OK")
```
Comment (rtso) [inline python/utils/transactions_processor.py:50]: Can you add more comment explaining what this does? How often do we send this response?

**Review by rtso — COMMENTED** [review]
How did you test this?

File: python/utils/transactions_processor.py (hunk 1 above)
Comment (larry-aptos) [reply python/utils/transactions_processor.py:53]: done, default is every 10 seconds from kubelet.

Comment (rtso) [reply python/utils/transactions_processor.py:53]: Thanks! Is this related to the 
```
indexer_grpc_http2_ping_interval_in_secs: 60
indexer_grpc_http2_ping_timeout_in_secs: 10
``` 
config vars that exist in the Rust config? If so we should add these to Python too.

File: python/utils/config.py [hunk 2]
```diff
@@ -13,6 +13,8 @@ class Config(BaseSettings):
     grpc_data_stream_endpoint: str
     grpc_data_stream_api_key: str
     db_connection_uri: str
+    health_port: int = 8080
```
Comment (rtso) [inline python/utils/config.py:16]: Can you also update `config.yaml.example`?

File: python/utils/transactions_processor.py (hunk 1 above)
Comment (larry-aptos) [reply python/utils/transactions_processor.py:53]: oh this is a different config; i can update it here as well.

Comment (rtso) [reply python/utils/transactions_processor.py:53]: What are those vars for?

Comment (larry-aptos) [reply python/utils/transactions_processor.py:53]: There are to make sure connection is alive from client side. Without these configs, clients trust the connections blindly, which can be a problem if the underlying network is not stable.

Comment (larry-aptos) [reply python/utils/transactions_processor.py:53]: done.

File: python/utils/config.py (hunk 2 above)
Comment (larry-aptos) [reply python/utils/config.py]: done

**Review by rtso — APPROVED** [review]

**Pull request merged**